### ISSUE TITLE
Output file format keyparser code duplicates

### ISSUE THREAD

**Issue by rijobro** [UCL/STIR#266]
In quite a few executables, we allow the user to give a parameter file for the saved image's format. 

In multiple places, we currently have code that reads the parameter file with `KeyParser`. It would be good to put this all into one place to avoid repetition.

**Comment by rijobro** [comment]
We have:
```
OutputFileFormat<e.g., DiscretisedDensity>::default_sptr()
```
So presumably, we could also have:
```
OutputFileFormat<e.g., DiscretisedDensity>::read_from_file(filename)
```

**Comment by KrisThielemans** [comment]
indeed, we can stick that functionality in there. Not sure about naming. Might be confusing to have an `OutputFileformat::read`....

By the way, we can probably lift this functionality to `ParsingObject`. Would then also be usable for other stuff such as post-filtering, reconstruction etc. Needs a bit of thought how to do that...
